app/helpers/decidim/after_sign_in_action_helper.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module Decidim
+  module AfterSignInActionHelper
+    extend ActiveSupport::Concern
+    include Decidim::FormFactory
+
+    included do
+      def default_url_options
+        url_options = {}
+        url_options[:locale] = current_locale unless current_locale == default_locale.to_s
+        url_options[:after_action] = request.params[:after_action] if request.params[:after_action].present?
+        url_options
+      end
+    end
+
+    def after_sign_in_action_for(user, action)
+      return if user.blank?
+      return unless action == "vote-initiative" && (scan = %r{/initiatives/i-(\d+)(\?.*)?}.match(read_stored_location_for(user)))
+
+      initiative = Decidim::Initiative.find(scan[1])
```

Comment:
* We should ensure `scan[1]` is a Number before finding in database
* initiative is on `user.organization`